src/hooks/useReports.ts

```diff
@@ -16,126 +17,93 @@ export interface ReportTransactionData {
+  const { data: epochs } = useEpochs();
+
+
+  const queryResults = useQuery({
     queryKey: ['reports', ownerId],
-    queryFn: async ({ pageParam }) => {
+    queryFn: async () => {
       if (
         !arIOReadSDK ||
-        startEpoch === undefined ||
+        epochs === undefined ||
         !gatewayStart ||
-        !observerAddress ||
-        pageParam === undefined
+        !observerAddress
       ) {
         throw new Error(
           'arIOReadSDK, startEpoch, ownerId, observerAddress, or gatewayStart not available',
         );
       }
 
-      let epochIndexToFetch = pageParam;
-
       let data: ReportTransactionData[] = [];
-      let completed = false;
-
-      const windowSize = 10;
-
-      // 10 epochs at a time and at least 10 data points, up until gateway start time
-      while (data.length < windowSize) {
-        // load ten epochs at a time
-        const batchedEpochIndicies = Array.from(
-          { length: windowSize },
-          (_, i) => epochIndexToFetch - i,
-        );
-        epochIndexToFetch -= windowSize;
-
-        const epochs = await Promise.all(
-          batchedEpochIndicies.filter(v => v >= 0).map((epochIndex) =>
-            arIOReadSDK.getEpoch({ epochIndex }),
-          ),
-        );
-
-        const filtered = epochs.filter(
-          (epoch) => epoch.endTimestamp >= gatewayStart,
-        );
-
-        const reportTransactionData = await Promise.all(
-          filtered.map((epoch) =>
-            arIOReadSDK.getObservations(epoch).then((observations) => {
-              const txid = observations.reports[observerAddress];
-
-              return txid
-                ? {
-                    txid,
-                    failedGateways: Object.values(
-                      observations.failureSummaries,
-                    ).reduce((acc, summary) => {
-                      return summary.includes(observerAddress) ? acc + 1 : acc;
-                    }, 0),
-                  }
-                : undefined;
-            }),
-          ),
-        );
 
-        const filteredTransactionData = reportTransactionData.reduce(
-          (acc, txData) =>
-            txData !== undefined ? { ...acc, [txData.txid]: txData } : acc,
-          {} as Record<string, { txid: string; failedGateways: number }>,
+      const reportTransactionData = await Promise.all(
+        epochs.map((epoch) =>
+        {
+          const observations = epoch?.observations;
+          const txid = observations?.reports[observerAddress];
+
+            return txid
+              ? {
+                  txid,
+                  failedGateways: Object.values(
+                    observations.failureSummaries,
+                  ).reduce((acc, summary) => {
+                    return summary.includes(observerAddress) ? acc + 1 : acc;
+                  }, 0),
+                }
+              : undefined;
+
+        })
+      );
+
+      const filteredTransactionData = reportTransactionData.reduce(
```

Comment:
This can be simplified further now that the first map is no longer doing network calls (hence the origina reason for the Promise.all()). Going to refactor this.